sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/StartingPoint.java

```diff
@@ -46,19 +45,19 @@ public StartingPoint(InitialPositionInStream position) {
 
   public StartingPoint(Instant timestamp) {
     this.timestamp = checkNotNull(timestamp, "timestamp");
-    this.position = null;
+    this.position = InitialPositionInStream.AT_TIMESTAMP;
```

Comment:
Why it was changed?

Reply:
Please take a look at for example [`org.apache.beam.sdk.io.kinesis.StartingPoint#getPositionName`](https://github.com/apache/beam/blob/d20e063bcd683880e311fd06430e6e2a0dd90937/sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/StartingPoint.java#L57). It had an `if` statement that could be replaced with setting `position` to `AT_TIMESTAMP` value in this constructor.
Another issue would be its use in [`org.apache.beam.sdk.io.kinesis.SimplifiedKinesisClient#buildShardFilterForStartingPoint`](https://github.com/apache/beam/blob/d20e063bcd683880e311fd06430e6e2a0dd90937/sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/StartingPoint.java#L57) would require `null` check.
It eliminates these problems.